### PR TITLE
feat(to-indexd): [TT-311] Moved to indexd completely

### PR DESCRIPTION
**Pull request by DrSpez**
THIS IS TO BE MERGED TO `release/fox` (changed to `release/versioning`, as it is a child of this branch)

### CODE REVIEW

File: sheepdog/transactions/upload/entity.py [hunk 1]
```diff
@@ -623,33 +630,6 @@ def register_index(self):
             record=alias, hashes=hashes, size=size, release='private'
         )
 
-    def flush_to_session(self):
```
Comment (DrSpez) [inline sheepdog/transactions/upload/entity.py]: That was a function redefininion, moved diff to the original function definition

File: sheepdog/transactions/upload/entity.py [hunk 2]
```diff
@@ -581,75 +581,6 @@ def get_metadata(self):
         metadata = {'acls': self.transaction.get_phsids()}
         return metadata
 
-    def register_index(self):
```
Comment (DrSpez) [inline sheepdog/transactions/upload/entity.py]: this is defined in `FileUploadEntity` as `_register_entity`

File: sheepdog/utils/transforms/__init__.py [hunk 3]
```diff
@@ -208,6 +208,14 @@ def convert_type(to_cls, key, value):
         if value is None:
             return None
 
+        # Currently, gdcdatamodel.models.File.__pg_properties__['file_size'] = (<type 'float'>, <type 'int'>, <type 'long'>)
+        # Though it needs to be <type 'int'> only as indexd allows only integer size
+        # Per Joe, The change of the model will require a full database migration and a maintanance shutdown
+        # Below is a sad temporary workaround:
+        if current_app.config.get('IS_GDC', True):
```
Comment (ProfOak) [inline sheepdog/utils/transforms/__init__.py]: Default to False

File: sheepdog/transactions/upload/sub_entities.py [hunk 4]
```diff
@@ -318,14 +278,16 @@ def _set_node_and_file_ids(self):
...
-                # use same id for both node entity id and file index
+            # generate entity_id if not provided
+            if not self.entity_id:
                 self.entity_id = str(uuid.uuid4())
-                self.file_index = self.entity_id
+            # use same id for both node entity id and file index
+            self.file_index = self.entity_id
         else:
+            # If hash uniqueness not enforced, do nothing
+            if not flask.current_app.config.get('ENFORCE_FILE_HASH_UNIQUENESS', True):
```
Comment (philloooo) [inline sheepdog/transactions/upload/sub_entities.py]: this config should be documented in dev_settings and describe the behavior is it's set to False

File: tests/submission/test_upload_signpost.py [hunk 5]
```diff
@@ -1,230 +0,0 @@
-"""
```
Comment (philloooo) [inline tests/submission/test_upload_signpost.py]: this file should be replaced with new test cases for scenario when ENFORCE_FILE_HASH_UNIQUENESS is False?

**Pull request merged**